contrib/src/main/java/com/datatorrent/contrib/hbase/HBasePOJOInputOperator.java

```diff
@@ -37,20 +37,21 @@
 import com.datatorrent.api.Context.OperatorContext;
 
 /**
+ * HBasePOJOInputOperator reads data from a HBase store, converts it to a POJO and puts it on the output port.
+ * The read from HBase is asynchronous.
  * @displayName HBase Input Operator
  * @category Input
  * @tags database, nosql, pojo, hbase
  * @since 3.1.0
  */
 @Evolving
-public class HBasePOJOInputOperator extends HBaseInputOperator<Object>
+public class HBasePOJOInputOperator extends HBaseScanOperator<Object>
 {
   private TableInfo<HBaseFieldInfo> tableInfo;
-  protected HBaseStore store;
   private String pojoTypeName;
-  private String startRow;
-  private String lastReadRow;
+  private Scan scan;
 
+  // Transients
   protected transient Class pojoType;
```

Comment:
Can you make pojoType, fieldValueGenerator & valueConverter private?
I don't see this being used outside of this class.